pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -174,93 +263,45 @@ func buildTopologyPairToScore(
-			terms := affinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-			if err := pm.processTerms(terms, pod, existingPod, existingPodNode, 1); err != nil {
-				return err
-			}
-		}
-		if hasAntiAffinityConstraints {
-			// For every soft pod anti-affinity term of <pod>, if <existingPod> matches the term,
-			// decrement <pm.counts> for every node in the cluster with the same <term.TopologyKey>
-			// value as that of <existingPod>`s node by the term`s weight.
-			terms := affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-			if err := pm.processTerms(terms, pod, existingPod, existingPodNode, -1); err != nil {
-				return err
-			}
+	}
+	if hasAntiAffinityConstraints {
+		if antiAffinityTerms, err = getProcessedTerms(pod, affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution); err != nil {
+			klog.Error(err)
+			return nil
 		}
+	}
 
-		if existingHasAffinityConstraints {
-			// For every hard pod affinity term of <existingPod>, if <pod> matches the term,
-			// increment <pm.counts> for every node in the cluster with the same <term.TopologyKey>
-			// value as that of <existingPod>'s node by the constant <ipa.hardPodAffinityWeight>
-			if hardPodAffinityWeight > 0 {
-				terms := existingPodAffinity.PodAffinity.RequiredDuringSchedulingIgnoredDuringExecution
-				// TODO: Uncomment this block when implement RequiredDuringSchedulingRequiredDuringExecution.
-				//if len(existingPodAffinity.PodAffinity.RequiredDuringSchedulingRequiredDuringExecution) != 0 {
-				//	terms = append(terms, existingPodAffinity.PodAffinity.RequiredDuringSchedulingRequiredDuringExecution...)
-				//}
-				for _, term := range terms {
-					if err := pm.processTerm(&term, existingPod, pod, existingPodNode, int64(hardPodAffinityWeight)); err != nil {
-						return err
-					}
-				}
-			}
-			// For every soft pod affinity term of <existingPod>, if <pod> matches the term,
-			// increment <pm.counts> for every node in the cluster with the same <term.TopologyKey>
-			// value as that of <existingPod>'s node by the term's weight.
-			terms := existingPodAffinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-			if err := pm.processTerms(terms, existingPod, pod, existingPodNode, 1); err != nil {
-				return err
-			}
-		}
-		if existingHasAntiAffinityConstraints {
-			// For every soft pod anti-affinity term of <existingPod>, if <pod> matches the term,
-			// decrement <pm.counts> for every node in the cluster with the same <term.TopologyKey>
-			// value as that of <existingPod>'s node by the term's weight.
-			terms := existingPodAffinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-			if err := pm.processTerms(terms, existingPod, pod, existingPodNode, -1); err != nil {
-				return err
-			}
-		}
-		return nil
+	pm := podAffinityPriorityMap{
+		topologyScore:         make(topologyPairToScore),
+		pod:                   pod,
```

Comment:
is it worth storing this guy? I feel like it should just be passed as parameter.

Reply:
we are computing this map from the perspective of this pod, so I think it is reasonable to have it. In fact passing it as a parameter means the functions of a specific ```podAffinityPriorityMap``` instance may accept different pods, which is not true.

Reply:
ok, so now we only have a single function relying on it, I removed it :)